src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -236,4 +240,72 @@ private static boolean checkIsAllLinesAreWhitespace(String[] lines, int lineFrom
         }
         return result;
     }
+
+    /**
+     *  Checks for empty LITERAL_CASE.
+     *
+     *  @param ast a {@code DetailAST} value
+     */
+    protected void checkForEmptyCase(DetailAST ast) {
+        final DetailAST slistToken = ast.getNextSibling();
+        if (slistToken != null
+                && slistToken.getType() == TokenTypes.SLIST) {
+            if (option == BlockOption.STMT) {
+                if (!isCaseHavingStmt(slistToken)) {
+                    log(ast.getLineNo(),
+                            ast.getColumnNo(),
+                            MSG_KEY_BLOCK_NO_STMT,
+                            ast.getText());
+                }
+            }
+            else if (!isCaseHavingText(slistToken)) {
+                log(ast.getLineNo(),
+                        ast.getColumnNo(),
+                        MSG_KEY_BLOCK_EMPTY,
+                        ast.getText());
+            }
+        }
+    }
+
+    /**
+     * Checks if case has any statements.
+     *
+     * @param slistAST a {@code DetailAST} value
+     * @return true if the case related to the SLIST token has atleast one statement
+     */
+    protected boolean isCaseHavingStmt(final DetailAST slistAST) {
+        DetailAST leftCurly;
+        for (leftCurly = slistAST.getFirstChild();
+             leftCurly != null; leftCurly = leftCurly.getNextSibling()) {
+            if (leftCurly.getFirstChild().getType() != TokenTypes.RCURLY) {
+                break;
+            }
+        }
+        return leftCurly != null;
+    }
+
+    /**
+     * Checks if case has any text.
+     *
+     * @param slistAST a {@code DetailAST} value
+     * @return true if the case related to the SLIST token has any text
+     */
+    protected boolean isCaseHavingText(final DetailAST slistAST) {
```

Comment:
`isCaseHavingText` is using the instance method `hasText` do you think it really is necessary ?